whisper.py

```diff
@@ -286,45 +297,57 @@ def __readHeader(fh):
   return info
 
 
-def setAggregationMethod(path, aggregationMethod, xFilesFactor=None):
-  """setAggregationMethod(path,aggregationMethod,xFilesFactor=None)
+def setXFilesFactor(path, xFilesFactor):
+  """Sets the xFilesFactor for file in path
+
+  path is a string pointing to a whisper file
+  xFilesFactor is a float between 0 and 1
+
+  returns the old xFilesFactor
+  """
 
-path is a string
-aggregationMethod specifies the method to use when propagating data (see ``whisper.aggregationMethods``)
-xFilesFactor specifies the fraction of data points in a propagation interval that must have known values for a propagation to occur.  If None, the existing xFilesFactor in path will not be changed
-"""
   with open(path, 'r+b', BUFFERING) as fh:
     if LOCK:
       fcntl.flock(fh.fileno(), fcntl.LOCK_EX)
 
-    packedMetadata = fh.read(metadataSize)
+    info = __readHeader(fh)
 
-    try:
-      (aggregationType, maxRetention, xff, archiveCount) = struct.unpack(metadataFormat, packedMetadata)
-    except (struct.error, ValueError):
-      raise CorruptWhisperFile("Unable to read header", fh.name)
+    __writeHeaderMetadata(fh, info['aggregationMethod'], info['maxRetention'],
+                          xFilesFactor, len(info['archives']))
 
-    try:
-      newAggregationType = struct.pack(longFormat, aggregationMethodToType[aggregationMethod])
-    except KeyError:
-      raise InvalidAggregationMethod("Unrecognized aggregation method: %s" %
-            aggregationMethod)
+    if AUTOFLUSH:
```

Comment:
setXFilesFactor and setAggregationMethod are basically copy+paste. I wasn't sure how to do this without changing the api and keeping it consistent.

I've got this now, still not sure if this is the right way:
https://github.com/piotr1212/whisper/blob/set_xff2/whisper.py#L300-L348
any suggestions?